lnrpc/walletrpc/walletkit_server.go

```diff
@@ -1414,12 +1431,45 @@ func (w *WalletKit) fundPsbtInternalWallet(account string,
 			}
 		}
 
+		// currentHeight is needed do determine whether the internal
+		// wallet utxo is still unconfirmed.
+		_, currentHeight, err := w.cfg.Chain.GetBestBlock()
+		if err != nil {
+			return fmt.Errorf("unable to retrieve current "+
+				"height: %v", err)
+		}
+
 		// We made sure the input from the user is as sane as possible.
 		// We can now ask the wallet to fund the TX. This will not yet
 		// lock any coins but might still change the wallet DB by
 		// generating a new change address.
 		changeIndex, err := w.cfg.Wallet.FundPsbt(
 			packet, minConfs, feeSatPerKW, account, keyScope,
+			func(utxo wtxmgr.Credit) bool {
+				// Wallet utxos which are unmined have a height
+				// of -1.
+				if utxo.Height != -1 ||
```

Comment:
should this not be `&&`? ie: height is <= current height AND not equal to -1?. 

Cause i think with this, the height could be `-1` meaning it is unconfirmed so the first condition would be false but then cause it is an `OR` check, it will go on to check the second condition which will be true

Reply:
Good catch 👌, my itest caught this behaviour